ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateUtil.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2018 Hyperwallet Systems Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.hyperwallet.android.ui.view.widget;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class is used for manage and convert date {@link DateWidget}
+ */
+class DateUtil {
+
+    private static final String TAG = DateUtil.class.getName();
+    private static final String SERVER_DATE_PATTERN = "yyyy-MM-dd";
+    private static final String WIDGET_DATE_PATTERN = "dd-MMM-yyyy";
+    private static final String DATE_PATTERN =
+            "(\\d{4})-(0?[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])";
+
+    private final SimpleDateFormat mServerDateFormat = new SimpleDateFormat(SERVER_DATE_PATTERN, Locale.getDefault());
+    private final SimpleDateFormat mWidgetDateFormat = new SimpleDateFormat(WIDGET_DATE_PATTERN, Locale.getDefault());
+
+    DateUtil() {
+    }
+
+    /**
+     * Convert Date from server format (yyyy-MM-dd) to DateWidget format (dd-MMM-yyyy)
+     *
+     * @param serverDate Date from server
+     * @return String date in the format (dd-MMM-yyyy)
```

Comment:
Please update comment to use `"dd MMMM yyyy"`

Reply:
Done